client/js/components/procedure/AdministrationImport/AdministrationImport.vue

```diff
@@ -134,48 +131,37 @@ export default {
       }
     },
 
-    /**
-     * Append a script tag to the head section which will be used to load a vue component dynamically
-     *
-     * @param {string} component
-     */
-    addComponentScript (component) {
-      const script = document.createElement('script')
-      script.id = component.name
-      script.type = 'text/javascript'
-      script.text = component.text
-
-      document.head.appendChild(script)
-      script.addEventListener('load', () => this.attachComponent(component))
-    },
-
-    /**
-     * Add component to this Vue wrapper component
-     */
-    addComponent (component) {
-      this.asyncComponents.push({
-        name: component.name,
-        permissions: component.permissions,
-        title: component.title
-      })
-    },
-
-    attachComponent (component) {
-      this.$options.components[component.name] = window[component.name]
-      const t = { ...component, component: this.$options.components[component.name] }
-      this.addComponent(t)
-    },
-
     loadComponents (hookName) {
       const params = {
         hookName: hookName
       }
 
-      dpRpc('addons.assets.load', params).then(response => {
-        this.addComponent(response)
-        this.addComponentScript(response)
+      dpRpc('addons.assets.load', params)
+        .then(response => checkResponse(response))
+        .then(response => {
+          const result = response[0].result
+
+          for (const key of Object.keys(result)) {
+            const addon = result[key]
+            const contentKey = addon.entry + '.umd.js'
+            const content = addon.content[contentKey]
+
+            /**
+             * The evaluation of the response content automatically binds the vue component
+             * to the window object. This way we can implement it in vue's internals to render
+             * the component.
+             */
+            eval(content)
+            this.$options.components[addon.entry] = window[addon.entry].default
+
+            this.asyncComponents.push({
```

Comment:
Why do we use `asyncComponents` here but not in `AddonWrapper`?

Reply:
The display the tabs in the import center we need the `asyncComponents` Object to be able to use the `<component :is="...` architecture of the tabs. The `AddonWrapper` has been crafted to mainly serve the purpose to be able to load addon components when the underlying structure is a twig template. Since the `AddonWrapper` is currently only enabled to load a single addon component we do not need the `asyncComponent` structure here because the addon component which is loaded will be automatically set as the component to be displayed.

Reply:
thanks for the explanation!